File: Java/0001-1000/001-100/61-70/66.java
class Solution {
    public int[] plusOne(int[] digits) {
        int c=1, n=digits.length, i;
        for(i=n-1;i>=0;i--) {
            digits[i] += c;
            c = digits[i]/10;
            digits[i] %= 10;
        }
        if(c==0) return digits;
        int[] digit = new int[n+1];
        digit[0] = 1;
        for(i=0;i<n;i++) {
            digit[i+1] = digits[i];
        }
        return digit;
    }
}

File: Python/0001-1000/001-100/61-70/66.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        i = len(digits) - 1
        l = []
        c = 1
        while i >= 0:
            s = digits[i] + c
            c = s//10
            s = s % 10
            l.append(s)
            i -= 1
        if c == 1:
            l.append(1)
        return l[::-1]
